service/src/main/java/de/adorsys/psd2/sandbox/xs2a/testdata/TestDataFileReader.java

```diff
@@ -0,0 +1,76 @@
+package de.adorsys.psd2.sandbox.xs2a.testdata;
+
+import de.adorsys.psd2.sandbox.xs2a.testdata.domain.Amount;
+import de.adorsys.psd2.sandbox.xs2a.testdata.domain.Transaction;
+import java.math.BigDecimal;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.time.LocalDate;
+import java.util.Currency;
+import java.util.HashMap;
+import java.util.List;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import lombok.NoArgsConstructor;
+import org.springframework.stereotype.Component;
+
+@Component
+@NoArgsConstructor
+final class TestDataFileReader {
+
+  private static final String FILE_NAME = "transactions_dump.txt";
+
+  static HashMap<String, Transaction> readTransactionsFromFile() {
+
+    List<String> txs;
+    try {
+      txs = Files.lines(Paths.get(ClassLoader.getSystemResource(FILE_NAME).toURI()))
+          .collect(Collectors.toList());
+    } catch (Exception e) {
+      throw new RuntimeException(e);
+    }
+
+    txs.remove(0);
+
+    List<Transaction> transactionList = txs.stream()
+        .map(TestDataFileReader::mapTransactionFromFile)
+        .collect(Collectors.toList());
+
+    return convertListToHashMap(transactionList);
+  }
+
+  private static Transaction mapTransactionFromFile(String transactionString) {
+    String[] transactionParts = transactionString.split("\\|");
+
+    return new Transaction(
+        UUID.randomUUID().toString(),
+        null,
+        new Amount(Currency.getInstance(transactionParts[6]), new BigDecimal(transactionParts[7])),
+        LocalDate.now(),
+        LocalDate.now(),
+        transactionParts[12],
+        transactionParts[13],
+        transactionParts[8],
+        transactionParts[9],
+        transactionParts[1],
+        transactionParts[2],
+        null,
+        transactionParts[3],
+        transactionParts[15],
+        transactionParts[16].trim().replaceAll(" +", " "),
```

Comment:
Why? This replace might deserve it's own method with a proper name.

Reply:
I removed it.